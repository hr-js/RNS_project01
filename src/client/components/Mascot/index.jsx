// @flow
import * as React from "react";
import styles from "./style.css";
// TODO: 画像ができたら差し替えする
import image from "../../images/test.png";

function Mascot(): React.Node {
  const { mascot, img } = styles;
  return (
    <div className={mascot}>
      <img src={image} alt="マスコット" className={img} />
    </div>
  );
}
export default Mascot;
