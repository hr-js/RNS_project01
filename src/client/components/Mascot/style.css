@charset "utf-8";
.mascot {
  position: relative;
  width: 150px;
}

.mascot .img {
  position: absolute;
  width: 110px;
  height: 200px;
  bottom: 0;
  left: 0;
}
