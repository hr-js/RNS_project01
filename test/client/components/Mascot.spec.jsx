import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Mascot from "@components/Mascot";

describe("Mascot.jsxのテスト", () => {
  describe("スナップショット", () => {
    it("正しいレンダリング", () => {
      const tree = renderer.create(<Mascot />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("コンポーネントのテスト", () => {
    const mascot = shallow(<Mascot />);
    const children = mascot.children();
    it("cssクラス: mascotをもつ", () => {
      expect(mascot.hasClass("mascot")).toBeTruthy();
    });

    it("子要素imgのcssクラス: imgをもつ", () => {
      expect(children.hasClass("img")).toBeTruthy();
    });

    it("子要素imgのaltの値は、マスコット", () => {
      expect(children.prop("alt")).toBe("マスコット");
    });

    it("子要素imgのsrcの値は、test-file-stub", () => {
      expect(children.prop("src")).toBe("test-file-stub");
    });
  });
});
