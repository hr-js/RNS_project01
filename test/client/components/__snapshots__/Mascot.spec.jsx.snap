// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mascot.jsxのテスト スナップショット 正しいレンダリング 1`] = `
<div
  className="mascot"
>
  <img
    alt="マスコット"
    className="img"
    src="test-file-stub"
  />
</div>
`;
